junit5-jupiter-starter-bazel/bazeljunit5/src/main/java/com/flexport/bazeljunit5/BazelJUnit5ConsoleLauncher.java

```diff
@@ -25,19 +29,45 @@
   private static final String SELECT_CLASS = "--select-class";
   private static final String SELECT_METHOD = "--select-method";
 
+  // LegacyXmlReportGeneratingListener with junit-jupiter generates this file in `--reports-dir` by
+  // default
+  // https://github.com/junit-team/junit5/blob/37e0f559277f0065f8057cc465a1e8eb91563af6/junit-platform-reporting/src/main/java/org/junit/platform/reporting/legacy/xml/LegacyXmlReportGeneratingListener.java#L116
+  private static final String DEFAULT_JUNIT_JUPITER_XML_OUTPUT_FILE = "TEST-junit-jupiter.xml";
+
   /** Transform args and invoke the real implementation. */
   public static void main(String... args) {
     int exitCode =
         ConsoleLauncher.execute(System.out, System.err, transformArgs(args)).getExitCode();
+    afterExecute(exitCode);
+
     System.exit(exitCode);
   }
 
+  /** Move the generated reports to where they should be. */
+  public static void afterExecute(int exitCode) {
+    fixXmlOutputFile(System.getenv("XML_OUTPUT_FILE"));
+  }
+
+  private static void fixXmlOutputFile(String xmlOutputFile) {
+    if (xmlOutputFile != null && !xmlOutputFile.isEmpty()) {
+      Path shouldPath = Paths.get(xmlOutputFile);
+      Path realPath = shouldPath.getParent().resolve(DEFAULT_JUNIT_JUPITER_XML_OUTPUT_FILE);
```

Comment:
I guess you'd need a command to merge multiple files when there are multiple test engines... 🤔

Reply:
Yes, I found this issue as well. Will have a try.

Reply:
We could add an option to the `ConsoleLauncher` to write a single xml report file to support this use case. Could you please open an issue for that on the junit5 repo?

Reply:
I did some tests, and find that no matter how many tests in one running, it will only generate a single XML file named like `TEST-*.xml`, so I think we can simply move it to `test.xml`. So it seems junit5 still does well. BTW, junit5 marks the XML report file as **legacy**: https://github.com/junit-team/junit5/blob/main/junit-platform-reporting/src/main/java/org/junit/platform/reporting/legacy/xml/LegacyXmlReportGeneratingListener.java
I am not familiar with junit5, and feedback is welcome.

Reply:
I found multiple XML files generated when I mixed both vintage tests and JUnit 5 Jupiter tests. 

Would be great to have the option to generate a single file as @marcphilipp suggwsted

Reply:
Hmm, interesting. Do you guys have a demo project for multiple XML files?